### PR TITLE
Added more rsync options and multi-server support

### LINKED ISSUE #3: Feature request: rsync options
Rsync has a lot of useful options, it would be nice to tune them in the config file. `usermap`, `groupmap` and `chown` are very much missing in terms of deployment process but looks like the best way is just to implement setting `raw` rsync options.

### PR DESCRIPTION
**Pull request by hryamzik**
Resolves #3 as well.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -29,12 +29,29 @@ resources:
...
 
+- name: sync-resource-multiple
+  type: rsync-resource
+  source:
+    servers:
+      - server1
+      - server2
+    base_dir: /sync_directory
+    user : user
+    disable_md5_path: false
```
Comment (mrsixw) [inline README.md]: Can you add descriptions for the options further up in the file please?

File: README.md [hunk 2]
```diff
@@ -29,12 +29,29 @@ resources:
     private_key: |
             ...
 
+- name: sync-resource-multiple
+  type: rsync-resource
+  source:
+    servers:
+      - server1
+      - server2
```
Comment (mrsixw) [inline README.md]: Also needs documentation stating that the first server in the list is the one used as the `check` and `in` origin

File: README.md (hunk 1 above)
Comment (mrsixw) [inline README.md]: This would be better named as disable version path (user doesn't necessarily need to know its MD5)

File: assets/out [hunk 3]
```diff
@@ -19,11 +19,21 @@ echo $MD5_TOOL 1>&2
 SCRIPT_INPUT='/tmp/input'
 cat > $SCRIPT_INPUT <&0 # STDIN params
 
-SERVER=$(jq -r '.source.server // ""' < $SCRIPT_INPUT)
+SERVERS=$(jq -r '.source.servers + [.source.server]|join(" ")'< $SCRIPT_INPUT)
 PORT=$(jq -r '.source.port // ""' < $SCRIPT_INPUT)
 BASE_DIR=$(jq -r '.source.base_dir // ""' < $SCRIPT_INPUT)
 USER=$(jq -r '.source.user // ""' < $SCRIPT_INPUT)
 SYNC_DIR=$(jq -r '.params.sync_dir // ""' < $SCRIPT_INPUT)
+RSYNC_OPTS_ARR=$(jq -r '.params.rsync_opts // ""' < $SCRIPT_INPUT)
```
Comment (mrsixw) [inline assets/out]: Why treat the rsync options and the disable MD5 (or version) differently?

Comment (hryamzik) [inline assets/out]: Because args are an array and MD5 is a single option.

**Comment by hryamzik** [comment]
Wow, I wasn't notified about these comments, going to respond and make changes ASAP.

**Comment by hryamzik** [comment]
@mrsixw I've pushed a new commit, please check it out.

**Comment by mrsixw** [comment]
Thanks for the changes. I've merged them. I've also rebuilt the docker image with this now and tagged it 'devel' whilst I test locally. If all good, I'll tag 'latest'

**Comment by hryamzik** [comment]
Perfect, thank you!

**Comment by mrsixw** [comment]
Tagged as latest now.

**Pull request merged**